articles/mysql-query-performance-optimization-tips/index.md

```diff
@@ -0,0 +1,180 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/mysql-query-performance-optimization-tips/
+title: MySQL Query Performance Optimization Tips
+description: This article will focus on the optimization of individual SQL statements and database structure as data volume grow.
+author: benson-kariuki
+date: 2020-09-29T00:00:00-13:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/mysql-query-performance-optimization-tips/hero.jpg
+    alt:
+---
+This article goes through common tips for optimizing MySQL queries. As data volume in your database grows, retrieving data from the database and other database operations become complex. This also requires more computing resources.
+<!--more-->
+### Introduction
+Most applications are database driven. Poorly designed SQL queries can significantly downgrade the performance of database-driven applications. According to this [MySQL developers guide](https://dev.mysql.com/doc/refman/8.0/en/optimize-overview.html), you can be proactive and plan for optimizations or troubleshoot queries and configurations after experiencing problems. This article will focus on the optimization of individual SQL statements and database structure.
+
+### Prerequisites
+To fully understand this article, prior knowledge of the MySQL database is essential. An understanding of different SQL queries and how they work is critical. This [MySQL tutorial](https://www.mysqltutorial.org/) and [tutorials point](https://www.tutorialspoint.com/mysql/index.htm) have free MySQL tutorials for beginners.
+
+### Benefits of MySQL Database Queries Optimization
+We optimize for speed and resources. Optimized queries can run faster and require less computing power.
+
+### Tips for MySQL Queries Optimization
+Fast queries are about response time. The goal is to have queries return the required result in the shortest time possible. How much time does a query take to execute? Most of the tools used to query MySQL database give details on time taken to run a query.
```

Comment:
"Most of the tools used to query MySQL database give..." should be "Most of the tools used to query a MySQL database give..." or "Most of the tools used to query MySQL databases give..."